server/asset/btc/btc.go

```diff
@@ -509,6 +514,173 @@ func (btc *Backend) VerifyUnspentCoin(_ context.Context, coinID []byte) error {
+	// P2WSH uses sha256, while P2SH uses ripemd160(sha256).
+	var expectedScriptHash []byte
+	if len(scriptHash) == 32 {
+		hash := sha256.Sum256(bondScript)
+		expectedScriptHash = hash[:]
+	} else {
+		expectedScriptHash = btcutil.Hash160(bondScript)
+	}
+	if !bytes.Equal(expectedScriptHash, scriptHash) {
+		err = fmt.Errorf("script hash check failed for output 0 of %s", msgTx.TxHash())
+		return
+	}
+
+	_, addrs, _, err := dexbtc.ExtractScriptData(bondOut.PkScript, chainParams)
+	if err != nil {
+		err = fmt.Errorf("error extracting addresses from bond output: %w", err)
+		return
+	}
+
+	txid := msgTx.TxHash()
+	bondCoinID = toCoinID(&txid, 0)
+	amt = bondOut.Value
+	bondAddr = addrs[0] // don't convert address, must match type we specified
+	lockTime = int64(lock)
+	bondPubKeyHash = pkh[:]
+
+	return
+}
+
+// BondVer returns the latest supported bond version.
+func (dcr *Backend) BondVer() uint16 {
+	return BondVersion
+}
+
+// ParseBondTx makes the package-level ParseBondTx pure function accessible via
+// a Backend instance. This performs basic validation of a serialized
+// time-locked fidelity bond transaction given the bond's P2SH redeem script.
+func (btc *Backend) ParseBondTx(ver uint16, rawTx []byte) (bondCoinID []byte, amt int64, bondAddr string,
+	bondPubKeyHash []byte, lockTime int64, acct account.AccountID, err error) {
+	return ParseBondTx(ver, rawTx, btc.chainParams)
+}
+
+// BondCoin locates a bond transaction output, validates the entire transaction,
+// and returns the amount, encoded lockTime and account ID, and the
+// confirmations of the transaction. It is a CoinNotFoundError if the
+// transaction output is spent.
+func (btc *Backend) BondCoin(ctx context.Context, ver uint16, coinID []byte) (amt, lockTime, confs int64, acct account.AccountID, err error) {
+	txHash, vout, errCoin := decodeCoinID(coinID)
+	if errCoin != nil {
+		err = fmt.Errorf("error decoding coin ID %x: %w", coinID, errCoin)
+		return
+	}
+
+	verboseTx, err := btc.node.GetRawTransactionVerbose(txHash)
+	if err != nil {
+		if isTxNotFoundErr(err) {
+			err = asset.CoinNotFoundError
+		}
+		return
+	}
+
+	if int(vout) > len(verboseTx.Vout)-1 {
+		err = fmt.Errorf("invalid output index for tx with %d outputs", len(verboseTx.Vout))
+		return
+	}
+
+	confs = int64(verboseTx.Confirmations)
+
+	rawTx, err := hex.DecodeString(verboseTx.Hex) // ParseBondTx will deserialize to msgTx, so just get the bytes
+	if err != nil {
+		err = fmt.Errorf("failed to decode transaction %s: %w", txHash, err)
+		return
+	}
+
+	txOut, err := btc.node.GetTxOut(txHash, vout, true) // check regular tree first
+	if err != nil {
+		if isTxNotFoundErr(err) { // should be txOut==nil, but checking anyway
+			err = asset.CoinNotFoundError
+			return
+		}
+		return
+	}
+	if txOut == nil { // spent == invalid bond
+		err = asset.CoinNotFoundError
+		return
+	}
+
+	_, amt, _, _, lockTime, acct, err = ParseBondTx(ver, rawTx, btc.chainParams)
+	return
```

Comment:
I feel like it should enforce segwit/non-segwit according to the `btc.segwit` flag.